store/tikv/lock_resolver.go

```diff
@@ -363,6 +363,14 @@ func (lr *LockResolver) resolveLocks(bo *Backoffer, callerStartTS uint64, locks
 
 			if status.primaryLock != nil && status.primaryLock.UseAsyncCommit && !exists {
 				err = lr.resolveLockAsync(bo, l, status)
+				// Treat nonAsyncCommitLock error as a soft error. It happens when a transaction
+				// just fallbacks from async commit, so a simple retry will work.
+				if _, ok := err.(*nonAsyncCommitLock); ok {
```

Comment:
Seems it never be true.
```go
	var errs []string
	for range regions {
		err1 := <-errChan
		if err1 != nil {
			errs = append(errs, err1.Error())
		}
	}

	if len(errs) > 0 {
		return nil, errors.Errorf("async commit recovery (sending CheckSecondaryLocks) finished with errors: %v", errs)
	}
```

Reply:
I should also add test to check resolve lock shouldn't return an error in such a scenario...